my_extension/src/enclave/enc_integer_avg/enc_integer_decrypt.rs

```diff
@@ -0,0 +1,38 @@
+use std::convert::TryInto;
+
+use aes::{cipher::generic_array::GenericArray, Aes128, BlockDecrypt, NewBlockCipher};
+
+use crate::host::enc_integer_avg::{DecodeError, EncInteger};
+
+use super::MASTER_KEY;
+
+pub(in crate::enclave) trait EncIntegerDecrypt {
+    fn decrypt(self) -> Result<i32, DecodeError>;
+}
+
+impl EncIntegerDecrypt for EncInteger {
+    fn decrypt(self) -> Result<i32, DecodeError> {
+        let encrypted = self.as_slice();
+
+        let key = GenericArray::from_slice(&MASTER_KEY);
+        let mut enc_block = GenericArray::clone_from_slice(&encrypted);
+
+        let cipher = Aes128::new(&key);
+
+        let dec_block = {
+            cipher.decrypt_block(&mut enc_block);
+            enc_block
+        };
+        let decrypted = dec_block.to_vec();
+        let decrypted: [u8; 16] = decrypted.try_into().map_err(|orig_vec: Vec<u8>| {
+            DecodeError::new(format!(
+                "base64-decoded data is {} bytes, while expected to be 4 bytes (4-byte integer with 16-byte padding))",
```

Comment:
あ、ここコメント間違ってる... base64もう使ってない

```suggestion
decrypted block data is {} bytes, while expected to be 16 bytes (4-byte integer with 16-byte padding)
```